### PR TITLE
Upgrade to Django 1.11.2

### PR DESCRIPTION
**Pull request by thallada**
In addition to upgrading Django, I also upgraded as many third party dependencies as I could to the latest versions in pip.

Most of the changes you see here are fixing backwards incompatible or newly deprecated Django patterns, or fixing new pylint warnings since I also updated pylint.

@edx/educator-dahlia

### CODE REVIEW

File: .travis/docker-compose-travis.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ services:
       ELASTICSEARCH_LEARNERS_UPDATE_INDEX: 'index_update'
     command: /edx/app/analytics_api/venvs/analytics_api/bin/python /edx/app/analytics_api/analytics_api/manage.py runserver 0.0.0.0:80
   insights:
-    image: edxops/insights:latest
+    image: edxops/insights:upgrade-django
```
Comment (thallada) [inline .travis/docker-compose-travis.yml:15]: This is required to have the Travis build succeed on this branch. Once this PR is merged into master, I will:

1. create a new docker image off the new master and push it as edxops/insights:latest
2. create a new PR to change this line back to `edxops/insights:latest`

File: requirements/production.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@
 -r base.txt
 
 MySQL-python==1.2.5		# GPL License
-PyYAML==3.11			# MIT License
+PyYAML==3.12			# MIT License
 gunicorn==0.17.4		# MIT
```
Comment (thallada) [inline requirements/production.txt:7]: I wanted to update gunicorn, but there are quite a few new major/minor versions and I didn't want to update it without a good way to test it. I believe only stage and prod environments use these dependencies.

File: acceptance_tests/course_validation/report_generators.py [hunk 3]
```diff
@@ -223,7 +223,7 @@ def generate_report(self):
                 path = 'performance/graded_content/{}'.format(slugify(assignment_type))
                 path = self.build_course_path(path)
                 status, elapsed = self.get_http_status_and_load_time(self.get_dashboard_url(path))
-                valid = (has_submissions and status == 200) or not has_submissions
+                valid = (status == 200 if has_submissions else not has_submissions)
```
Comment (efischer19) [inline acceptance_tests/course_validation/report_generators.py]: I think `(status == 200 if has_submissions else True)` would be a little clearer here

**Review by efischer19 — APPROVED** [review]
1 minor suggestion, LGTM otherwise

🚢

File: requirements/base.txt [hunk 4]
```diff
@@ -1,29 +1,32 @@
...
 djangorestframework-csv==2.0.0 # BSD
 # Dependency of djangorestframework
-django-crispy-forms==1.6.0  # MIT
+django-crispy-forms==1.6.1  # MIT
 django-soapbox==1.3         # BSD
-django-waffle==0.11.1		# BSD
+# django-waffle==0.11.1		# BSD
+# FIXME: django-waffle management commands are broken on Django 1.10+. This repo is not the official django-waffle repo.
+# Check back with django-waffle later to see if it gets fixed.
+git+https://github.com/jdavidagudelo/django-waffle.git@349d43eea7e10ccb995c8987f0c4bd8a7a240b55#egg=django-waffle
```
Comment (dsjen) [inline requirements/base.txt:18]: Thanks for digging that one up!

**Review by dsjen — APPROVED** [review]
This is great!  I tested it out and it is working like a charm.  Maybe hold off on merging until the release later today?

**Comment by thallada** [comment]
I'm going to merge this now and update edxops/insights:latest. Look out for the follow-up PR coming soon.

**Pull request merged**